redaxo/src/core/lib/rex.php

```diff
@@ -316,6 +316,27 @@ public static function getConsole()
         return self::getProperty('console', null);
     }
 
+    /**
+     * @param int $db
+     * @psalm-param positive-int $db
+     *
+     * @throws rex_exception
+     *
+     * @return rex_config_db
+     */
+    public static function getDbConfig($db = 1)
+    {
+        $config = self::getProperty('db', null);
+
+        if (!$config) {
+            $configFile = rex_path::coreData('config.yml');
+
+            throw new rex_exception('Unable to read db config from config.yml "'. $configFile .'"');
+        }
```

Comment:
Fehlt hier in dem if das yaml parsing, oder ist es ein überbleibsel, und das if sollte eigentlich ganz weg?

Reply:
aus meiner sicht kann es ganz weg. die yaml oder der cache werden immer in der boot.php geladen. daher sehe ich eigentlich keine situation, wo es notwendig ist.

Reply:
ich habe es bewusst weggelassen da ich nicht parsen will, da es immer da sein sollte.
das hier ist nur eine versicherung (die ich ohnehin brauche, weil die SA sonst meckert wg. $config could be null

Reply:
Ach ok, jetzt verstehe ich es. Hatte hier irgendwie falsch gedacht.